### ISSUE TITLE
test Task

### ISSUE THREAD

**Issue by maiermic** [deadfoxygrandpa/Elm-Test#26]
I'd like to test a Task. Is this already possible?

**Comment by deadfoxygrandpa** [comment]
No, it’s not possible right now. This will require some planning, I think,
since Tasks can perform effects. Are you thinking more like having a Task x
a, performing it, and making assertions against the ultimate result, or are
you thinking more along the lines of testing its actual effect?
​

**Comment by maiermic** [comment]
I work on a native package that uses Tasks to calculate the result of asynchronous functions. I'm mainly interested in making assertions against the ultimate result. But I also like to test if errors (effects) appear as expected.

**Comment by deadfoxygrandpa** [comment]
This is something I'd like to add in the medium term. I'm not yet sure of the best way to go about it, I have some ideas, but it will surely involve adding Native code, which is a somewhat significant amount of work.

**Comment by langston-barrett** [comment]
I would love to see this feature! Tasks are the hardest part of Elm to debug. [You can't evaluate them at the REPL](https://github.com/elm-lang/elm-repl/issues/77) or in test suites rn :cry:
